backend/layers/business/business.py

```diff
@@ -692,6 +693,20 @@ def tombstone_collection(self, collection_id: CollectionId) -> None:
         self.delete_all_dataset_versions_from_public_bucket_for_collection(collection_id)
         self.database_provider.tombstone_collection(collection_id)
 
+    def resurrect_collection(self, collection_id: CollectionId) -> None:
+        """
+        Resurrect a tombstoned Collection (untombstone Collection and un-delete public s3 assets). For admin use only.
+        """
+        collection = self.get_canonical_collection(collection_id)
+        if not collection.tombstoned:
+            raise CollectionIsNotTombstonedException()
```

Comment:
avoid negative logic, and add a test for this error case.
```suggestion
            raise CollectionIsAliveException()
```

Reply:
How about `CollectionIsPublicException` 😄  as much as I like the commitment to 🧟 💀 vibes...